packages/react-components/react-aria/src/AriaLiveAnnouncer/useAriaNotifyAnnounce.ts

```diff
@@ -0,0 +1,46 @@
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import * as React from 'react';
+
+import type { AriaLiveAnnounceFn } from './AriaLiveAnnouncer.types';
+
+type AriaNotifyOptions = {
+  notificationID?: string;
+  priority?: 'none' | 'important';
+  interrupt?: 'all' | 'pending' | 'none';
+};
+
+type DocumentWithAriaNotify = Document & {
+  ariaNotify: (message: string, options: AriaNotifyOptions) => void;
+};
+
+/* INTERNAL: implementation of the announcer using the ariaNotify API */
+export const useAriaNotifyAnnounce_unstable = (): AriaLiveAnnounceFn => {
+  const { targetDocument } = useFluent();
+
+  const announce: AriaLiveAnnounceFn = React.useMemo(
+    () =>
+      (message, options = {}) => {
+        if (!targetDocument) return;
+
+        let { alert = false, polite, priority, batchId } = options;
+
+        // default priority to 0 if polite, 2 if alert, and 1 by default
+        // used to set both ariaNotify's priority and interrupt
+        if (typeof priority !== 'number') {
```

Comment:
I think you can make `priority` default to `1` where you destructure the options and eliminate this `if` (you'll still need the internal conditional but it can become simpler).

Reply:
Your example is pretty much what I was thinking but I think you can collapse it into a single line:

```ts
// Default value at destructure
let { priority = 1, polite, /* ... other props here ... */ } = props;
// ...
priority = polite ? 0 : alert ? 2 : priority;
```

Reply:
ah, the one thing is I was thinking `props.priority` should override everything else

Reply:
I see ya, that makes sense.